dec dev update (#117)
irods/irods.org: content/posts/irods-development-update-december-2025.md

Title: iRODS Development Update: December 2025
Date: 2025-12-10 10:00
Author: Kory Draughn
Slug: irods-development-update-december-2025
Status: published

Hello reader,

> We just held our fourth and final [TRiRODS](https://irods.org/trirods) meeting of the year. Terrell Russell, Executive Director of the iRODS Consortium, gave a talk about using open-source tools to gather and present various metrics derived from the iRODS catalog. The [video](https://www.youtube.com/watch?v=IWKrd5b-57Q) is now available at YouTube.

Reviewer: didn't have may.

```suggestion
We just held our final [TRiRODS](https://irods.org/trirods) meeting of the year. Terrell Russell, Executive Director of the iRODS Consortium, gave a talk about using open-source tools to gather and present various metrics derived from the iRODS catalog. The [video](https://www.youtube.com/watch?v=IWKrd5b-57Q) is now available at YouTube.
```

Author: Done.

For our final development update, work on a new authentication scheme (called **irods**) has landed in upstream. It supports password hashing, session tokens, and various commands for managing user credentials. Support for the CRC64/NVME checksum algorithm has been merged too.

Here's a big one. We're preparing for an initial release of the Policy Composition framework! The code has been updated to address memory issues and we're now reviewing documentation to make sure users have what they need.

> Testing of the Python iRODS client (PRC) is now automated via GitHub Actions. This is huge because it speeds up the release process. The current plan is to do the same for other repositories.

Reviewer: ```suggestion
Testing of the Python iRODS client (PRC) is now automated via GitHub Actions. This is huge because it will speed up the release process. The current plan is to do the same for other repositories.
```

irods4j 0.6.0 is now available. This release includes support for JSSE TrustManagers. Just like the PRC, we've fully automated all testing of the library using GitHub Actions.

> For Cyberduck, our efforts to replace Jargon with irods4j are coming to a close. Tests are passing and the primary PR has been approved. We're now wrapping up documentation. Keep in mind that this work does raise the version requirement of iRODS to 4.3.2.

Reviewer: ```suggestion
For Cyberduck, our efforts to replace Jargon with irods4j are coming to a close. Tests are passing and the primary PR has been approved. We're now wrapping up documentation. Keep in mind that this work raises the required iRODS server version to 4.3.2.
```
